### PR TITLE
26 implement aws ssm connection type

### PR DESCRIPTION
**Pull request by dominik-przybyl-wttech**

### CODE REVIEW

File: internal/client/connection_aws_ssm.go [hunk 1]
```diff
@@ -1,38 +1,136 @@
...
+func (a *AWSSSMConnection) CopyFile(localPath string, remotePath string) error {
+	// Upload file to the remote instance using SSM Parameter Store
+	fileContent, err := os.ReadFile(localPath)
+	if err != nil {
+		return fmt.Errorf("ssm: error reading local file: %v", err)
+	}
+
+	putParameterInput := &ssm.PutParameterInput{
+		Name:      aws.String(remotePath),
+		Value:     aws.String(string(fileContent)),
```
Comment (krystian-panek-wttech) [inline internal/client/connection_aws_ssm.go:125]: how about 500 mb string ?:D

File: internal/client/connection_aws_ssm.go [hunk 2]
```diff
@@ -1,38 +1,120 @@
...
 	Region     string
+	Client     *ssm.Client
+	SessionId  *string
 }
 
-func (A AWSSSMConnection) Info() string {
-	//TODO implement me
-	panic("implement me")
+func (a *AWSSSMConnection) Info() string {
+	return fmt.Sprintf("ssm: instance='%s', region='%s'", a.InstanceID, a.Region)
```
Comment (krystian-panek-wttech) [inline internal/client/connection_aws_ssm.go:23]: instance_id

File: internal/client/connection_aws_ssm.go [hunk 3]
```diff
@@ -1,38 +1,120 @@
...
+func (a *AWSSSMConnection) User() string {
+	return "root"
 }
 
-func (A AWSSSMConnection) Connect() error {
-	//TODO implement me
-	panic("implement me")
+func (a *AWSSSMConnection) Connect() error {
+	// Specify the AWS region
+	cfg, err := config.LoadDefaultConfig(context.Background(), config.WithRegion(a.Region))
```
Comment (krystian-panek-wttech) [inline internal/client/connection_aws_ssm.go:32]: whole connection should use same context

File: internal/client/connection_aws_ssm.go [hunk 4]
```diff
@@ -1,38 +1,120 @@
...
+		return err
+	}
+
+	// Create an SSM client
+	client := ssm.NewFromConfig(cfg)
+	if err != nil {
+		return fmt.Errorf("ssm: error creating AWS session: %v", err)
+	}
+
+	startSessionInput := &ssm.StartSessionInput{
```
Comment (krystian-panek-wttech) [inline internal/client/connection_aws_ssm.go:43]: 3 lines to 1

File: internal/provider/instance_client.go [hunk 5]
```diff
@@ -220,7 +220,7 @@ func (ic *InstanceClient) ReadStatus() (InstanceStatus, error) {
 
 func (ic *InstanceClient) bootstrap() error {
 	return ic.doActionOnce("bootstrap", ic.cl.WorkDir, func() error {
-		return ic.runScript("bootstrap", ic.data.System.Bootstrap, ".")
+		return ic.runScript("bootstrap", ic.data.System.Bootstrap, ic.cl.WorkDir)
```
Comment (krystian-panek-wttech) [inline internal/provider/instance_client.go:223]: why?

**Review by krystian-panek-wttech — APPROVED** [review]

File: internal/provider/instance_client.go (hunk 5 above)
Comment (bd-dominik-przybyl) [reply internal/provider/instance_client.go:223]: some commands downloads temporary files, this is to ensure that it is done in dedicated directory

**Pull request merged**